packages/blocks/src/store/reducer.js

```diff
@@ -52,6 +57,72 @@ function getUniqueItemsByName( items ) {
 	}, [] );
 }
 
+function bootstrappedBlockTypes( state = {}, action ) {
+	switch ( action.type ) {
+		case 'ADD_BOOTSTRAPPED_BLOCK_TYPE':
+			const { name, blockType } = action;
+			const serverDefinition = state[ name ];
+			let newDefinition;
+			// Don't overwrite if already set. It covers the case when metadata
+			// was initialized from the server.
+			if ( serverDefinition ) {
+				// The `selectors` prop is not yet included in the server provided
+				// definitions and needs to be polyfilled. This can be removed when the
+				// minimum supported WordPress is >= 6.3.
+				if (
+					serverDefinition.selectors === undefined &&
+					blockType.selectors
+				) {
+					newDefinition = {
+						...serverDefinition,
+						selectors: blockType.selectors,
+					};
+				}
+
+				// The `autoInsert` prop is not yet included in the server provided
+				// definitions and needs to be polyfilled. This can be removed when the
+				// minimum supported WordPress is >= 6.4.
```

Comment:
Added this comment, and it implements the suggestion I shared with @ockham in https://github.com/WordPress/gutenberg/pull/52969/files#r1310187507.